test/check-machines-migrate

```diff
@@ -0,0 +1,187 @@
+        else:
+            # Older libvirt needs the destination storage preallocated
+            if self.machine.image in ["ubuntu-2004"]:
+                machine2.execute("qemu-img create -f qcow2 /var/lib/libvirt/images/subVmTest1-2.img 50M")
+
+        self.waitVmRow("subVmTest1")
+        self.goToVmPage("subVmTest1")
+
+        self.performAction("subVmTest1", "migrate")
+        b.wait_visible("#migrate-modal")
+
+        b.set_input_text("#dest-uri-input", "qemu+ssh://root@machine2/system")
+
+        b.set_checked("#storage #copy", copy_storage)
+
+        b.set_checked("#live", live)
+        if not live:
+            b.wait_not_present("#temporary")
+        else:
+            b.wait_visible("#temporary")
+            b.set_checked("#temporary", temporary)
+            if temporary:
+                b.wait_visible(".pf-c-modal-box__footer .pf-c-alert")
+
+        # The VM does not exist in destination before migration happens
+        self.assertNotIn("subVmTest1", machine2.execute("virsh list --all"))
+
+        b.click("#migrate-button")
+        b.wait_not_present("#migrate-modal")
+
+        # Verify the state of the migrated VM in the destination host
+        # The migrated VM should always exist on destination - domstate depends on the options of the dialog set
+        wait(lambda: "subVmTest1" in machine2.execute("virsh list --all"), delay=3)
+        if live:
+            wait(lambda: "running" in machine2.execute("virsh domstate subVmTest1"), delay=3)
+            if temporary:
+                # VM on the destination host should be transient
+                self.assertIn("Persistent:     no", machine2.execute("virsh dominfo subVmTest1"))
+            else:
+                # VM on the destination host should be transient
+                self.assertIn("Persistent:     yes", machine2.execute("virsh dominfo subVmTest1"))
+        else:
+            # Non-live migration will migrate only offline VM config and leave the VM running on the source host
+            self.assertIn("shut off", machine2.execute("virsh domstate subVmTest1"))
+
+        # Verify the state of the migrated VM in the source host
+        if live:
+            if temporary:
+                # VM on the source host should be still defined
+                self.assertIn("shut off", machine1.execute("virsh domstate subVmTest1"))
+            else:
+                # VM should not exist on the source host
+                self.assertNotIn("subVmTest1", machine1.execute("virsh list --all"))
+        else:
+            # Non-live migration will migrate only offline VM config and leave the VM running on the source host
+            self.assertIn("running", machine1.execute("virsh domstate subVmTest1"))
+
+        if not copy_storage:
+            # Running VM on destination host has disk with a block storage which is available on both source and destination
+            self.assertIn("disk type='block'", machine2.execute("virsh dumpxml subVmTest1"))
```

Comment:
I think that needs to be more specific. I suppose that's looking for the RAM disk, but wouldn't the VM's root file system also have this?

Reply:
no, the subVmTest1 default root disk (and only disk) is `type='file'`. The disk for sharedStorage test is `type='block'`